### ISSUE TITLE
DSBulk fails to load blob values

### ISSUE THREAD

**Issue by jaibheem** [datastax/dsbulk#383]
Hello Team,

I am trying to unload and load blob values from one cluster to another cluster using dsbulk: 1.7.0 but I see the issue during the import.

after the import when I check the data fro cqlsh I see the below errors.

     tid      | val
    ---------------
     abc12347 |       '\xac\xed\x00\x05sr\x00\x10java.lang.Double\x80\xb3\xc2J)k\xfb\x04\x02\x00\x01D\x00\x05valuexr\x00\x10java.lang.Number\x86\xac\x95\x1d\x0b\x94\xe0\x8b\x02\x00\x00xpBwtj\xf0#`\x00'
     abc12346 | '\xac\xed\x00\x05sr\x00\x10java.lang.Double\x80\xb3\xc2J)k\xfb\x04\x02\x00\x01D\x00\x05valuexr\x00\x10java.lang.Number\x86\xac\x95\x1d\x0b\x94\xe0\x8b\x02\x00\x00xpBwtj\xed\x90\x80\x00'
     abc12345 | '\xac\xed\x00\x05sr\x00\x10java.lang.Double\x80\xb3\xc2J)k\xfb\x04\x02\x00\x01D\x00\x05valuexr\x00\x10java.lang.Number\x86\xac\x95\x1d\x0b\x94\xe0\x8b\x02\x00\x00xpBwtj\xef\x8a\x80\x00'
    
    (3 rows)
    Failed to format value '\xac\xed\x00\x05sr\x00\x10java.lang.Double\x80\xb3\xc2J)k\xfb\x04\x02\x00\x01D\x00\x05valuexr\x00\x10java.lang.Number\x86\xac\x95\x1d\x0b\x94\xe0\x8b\x02\x00\x00xpBwtj\xf0#`\x00' : 'ascii' codec can't decode byte 0xac in position 0: ordinal not in range(128)
    Failed to format value '\xac\xed\x00\x05sr\x00\x10java.lang.Double\x80\xb3\xc2J)k\xfb\x04\x02\x00\x01D\x00\x05valuexr\x00\x10java.lang.Number\x86\xac\x95\x1d\x0b\x94\xe0\x8b\x02\x00\x00xpBwtj\xed\x90\x80\x00' : 'ascii' codec can't decode byte 0xac in position 0: ordinal not in range(128)
    1 more decoding errors suppressed.I also tried to set `codec.binary` but it didn't help https://docs.datastax.com/en/dsbulk/doc/dsbulk/reference/codecOptions.html#codecOptions__dsbulkCodecOptionsBinary

┆Issue is synchronized with this [Jira Task](https://datastax-oss.atlassian.net/browse/BULK-5) by [Unito](https://www.unito.io)

**Comment by adutra** [comment]
Hi thanks for reporting this.

Please note that DSBulk is not compatible with CQLSH COPY. Each tool formats and parses CSV data differently and you should never expect that the output produced by one tool will work if used as the input for the other tool.

Would you mind giving me a full example to reproduce the problem? The snippet you gave is not clear enough as I don't understand if this is expected to be a CSV file or a table content or something else.

Also, you might have some schema issues as well: your snippet mentions "'ascii' codec" but binary data cannot be stored in ascii columns, you should use blob columns instead.

Good luck!

**Comment by adutra** [comment]
@jaibheem were you able to fix your issue?

**Comment by adutra** [comment]
Closing as we didn't hear back from the reporter.

**Comment by jaibheem** [comment]
sorry for the late response. finally I was able to figure out the issue, posting it here so it can be useful for others. DSBulk was able properly unload and load the blob data. I confirmed that by testing my application pointing both the clusters. However the issue I reported is mainly a Python bug. Earlier I was using a cqlsh from python-pip and later I switched to the correct version of Cqlsh that is distributed from the cassandra tar and the output looked fine.
